packages/shared/lib/services/account.service.ts

```diff
@@ -84,15 +88,19 @@ class AccountService {
      * @desc create a new account and assign to the default environments
      */
     async createAccount(name: string): Promise<DBTeam | null> {
+        // TODO: use transaction
         const result = await db.knex.from<DBTeam>(`_nango_accounts`).insert({ name }).returning('*');
 
-        if (result[0]?.id) {
-            await environmentService.createDefaultEnvironments(result[0].id);
+        if (!result[0]) {
+            return null;
+        }
 
-            return result[0];
+        await environmentService.createDefaultEnvironments(result[0].id);
+        if (flagHasPlan) {
```

Comment:
why the flag since the plans table is not being used yet

Reply:
it's for enterprise and self-hosted
but the plans table will be used on cloud after this PR